### PR TITLE
Support serializing to the PKCS#12 file format.

### PR DESCRIPTION
**Pull request by boumenot**

### CODE REVIEW

File: packer/builder/azure/pkcs12/pbkdf_test.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package pkcs12
+
+import (
+	"bytes"
+	"testing"
+)
+
+var hit = false
```
Comment (paulmey) [inline packer/builder/azure/pkcs12/pbkdf_test.go]: not sure what this is for...

Comment (boumenot) [inline packer/builder/azure/pkcs12/pbkdf_test.go]: I don't know either, it was from the original code.  I'm removed it.

**Comment by paulmey** [comment]
Can we remove everything that is not necessary for encoding? We should be using the x/crypto package for decoding.

**Comment by boumenot** [comment]
I removed all of the top level decoding pieces.  The lower level decoding pieces are useful for unit testing because it's easier to assert that something round trips rather than it just encrypts.  I moved all decoding routines into the _test files to make the intent clearer.

**Pull request merged**